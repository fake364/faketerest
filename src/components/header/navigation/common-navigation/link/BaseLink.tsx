import React from 'react';
import Link, { LinkProps } from 'next/link';

interface Props extends LinkProps {
	className?: string;
}

const BaseLink: React.FC<Props> = ({ children, className, ...rest }) => {
	return (
		<Link {...rest}>
			<span
				className={`${
					className ? className : ''
				} text-black cursor-pointer hover:underline font-medium`}
			>
				{children}
			</span>
		</Link>
	);
};

export default BaseLink;
