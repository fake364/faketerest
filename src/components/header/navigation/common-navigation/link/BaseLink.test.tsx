import React from 'react';
import { mount } from 'enzyme';
import BaseLink from './BaseLink';

describe('BaseLink component', () => {
	it('should render properly with default props', () => {
		const component = mount(
			<BaseLink href={'/test'}>Test component</BaseLink>
		);
		expect(component).toMatchSnapshot();
	});

	it('should render properly with classname', () => {
		const component = mount(
			<BaseLink className="test-class" href={'/test'}>
				Test component
			</BaseLink>
		);
		expect(component).toMatchSnapshot();
	});
});
