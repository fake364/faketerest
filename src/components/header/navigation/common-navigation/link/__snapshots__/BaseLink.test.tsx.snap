// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLink component should render properly with classname 1`] = `
<BaseLink
  className="test-class"
  href="/test"
>
  <ForwardRef(LinkComponent)
    href="/test"
  >
    <span
      className="test-class text-black cursor-pointer hover:underline font-medium"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Test component
    </span>
  </ForwardRef(LinkComponent)>
</BaseLink>
`;

exports[`BaseLink component should render properly with default props 1`] = `
<BaseLink
  href="/test"
>
  <ForwardRef(LinkComponent)
    href="/test"
  >
    <span
      className=" text-black cursor-pointer hover:underline font-medium"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Test component
    </span>
  </ForwardRef(LinkComponent)>
</BaseLink>
`;
